listenstore/listenstore/listenstore.py

```diff
@@ -106,6 +106,13 @@ def insert_postgresql(self, listens):
                     self.log.error(e)
 
     def fetch_listens_from_storage(self, user_id, from_ts, to_ts, limit, order, precision):
+        """ The timestamps are stored as UTC in the postgres datebase while on retrieving
+            the value they are converted to the local server's timezone. So to compare
+            datetime object we need to create a object in the same timezone as the server.
```

Comment:
I tested it on my PC and on checking the output of the poastgres query from python i saw it had a `tzinfo` field (it was not None). And then loading using `utcfromtimestamp()` You can see the bug that the previous button on the user profile page will not vanish. The exact problem was, i was comparing the the created UTC datetime object against the postgres object. But postgres outputs the datetime in the local server's timezone (from the postgres documentation: https://ptpb.pw/bHsZ). There was again 2 ways that i could fix this,

**Solution 1:**
![1](https://cloud.githubusercontent.com/assets/14854424/17065656/b2e50cb8-5060-11e6-9014-e6f921efa491.png)

**Solution 2:**
![2](https://cloud.githubusercontent.com/assets/14854424/17065660/b6dd0e56-5060-11e6-969a-37249e876e02.png)